xchange-ftx/src/main/java/org/knowm/xchange/ftx/service/FtxLendingServiceRaw.java

```diff
@@ -0,0 +1,197 @@
+    }
+  }
+
+  public List<FtxLendingHistoryDto> histories(String subaccount) {
+    try {
+      return ftx.getlendingHistories(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount
+      ).getResult();
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't get lending infos subAccount: " + subaccount, e);
+    }
+  }
+
+  public List<FtxLendingHistoryDto> histories(String subaccount, List<String> coins) {
+    Objects.requireNonNull(coins);
+    return histories(subaccount).stream()
+        .filter(lendingHistory -> coins.contains(lendingHistory.getCoin()))
+        .collect(Collectors.toList());
+  }
+
+  public FtxLendingHistoryDto history(String subaccount, String coin) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin))
+      throw new FtxLendingServiceException("Coin are blank or empty");
+    return histories(subaccount).stream()
+        .filter(lendingHistory -> lendingHistory.getCoin().equalsIgnoreCase(coin))
+        .findFirst()
+        .orElse(null);
+  }
+
+  public List<FtxLendingInfoDto> infos(String subaccount) {
+    try {
+      return ftx.getLendingInfos(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator,
+          subaccount
+      ).getResult();
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't get lending infos subAccount: " + subaccount, e);
+    }
+  }
+
+  public List<FtxLendingInfoDto> infos(String subaccount, List<String> coins) {
+    Objects.requireNonNull(coins);
+    return infos(subaccount).stream()
+        .filter(lendingInfo -> coins.contains(lendingInfo.getCoin()))
+        .collect(Collectors.toList());
+  }
+
+  public FtxLendingInfoDto info(String subaccount, String coin) {
+    Objects.requireNonNull(coin);
+    if (StringUtils.isNotBlank(coin))
+      throw new FtxLendingServiceException("Coin are blank or empty");
+    return infos(subaccount).stream()
+        .filter(lendingInfo -> lendingInfo.getCoin().equalsIgnoreCase(coin))
+        .findFirst()
+        .orElse(null);
+  }
+
+  public List<FtxLendingRatesDto> rates() {
+    try {
+      return ftx.getLendingRates(
+          exchange.getExchangeSpecification().getApiKey(),
+          exchange.getNonceFactory().createValue(),
+          signatureCreator
+      ).getResult();
+    } catch (IOException e) {
+      throw new FtxLendingServiceException("Can't get lending rates", e);
+    }
+  }
+
+  public List<FtxLendingRatesDto> rates(List<String> coins) {
+    Objects.requireNonNull(coins);
+    return rates().stream()
+        .filter(lendingRates -> coins.contains(lendingRates.getCoin()))
+        .collect(Collectors.toList());
+  }
+
+  public FtxLendingRatesDto rate(String coin) {
```

Comment:
can we just get rid of this method and have users who want to fund out rate for a single coin just pass in singleton list in the method above?

Reply:
Why I think I better like that